projects/packages/identity-crisis/src/class-identity-crisis.php

```diff
@@ -432,14 +432,59 @@ public static function get_sync_error_idc_option( $response = array() ) {
 	 * @return bool
 	 * @since 0.2.0
 	 * @since-jetpack 4.3.2
+	 * @deprecated $$next-version$$ Use should_handle_idc()
+	 * @see Automattic\Jetpack\Identity_Crisis::should_handle_idc
 	 */
 	public static function sync_idc_optin() {
-		if ( Constants::is_defined( 'JETPACK_SYNC_IDC_OPTIN' ) ) {
+		_deprecated_function( __METHOD__, '0.2.6', 'Automattic\\Jetpack\\Identity_Crisis::should_handle_idc' );
+		return self::should_handle_idc();
+	}
+
+	/**
+	 * Returns the value of the jetpack_should_handle_idc filter or constant.
+	 * If set to true, the site will be put into staging mode.
+	 *
+	 * This method uses both the current jetpack_should_handle_idc filter and constant and the
+	 * legacy jetpack_sync_idc_optin filter and constant to determine whether an IDC should be
+	 * handled.
+	 *
+	 * @return bool
+	 * @since $$next-version$$
+	 */
+	public static function should_handle_idc() {
+		if ( Constants::is_defined( 'JETPACK_SHOULD_HANDLE_IDC' ) ) {
```

Comment:
I'm not sure if this new constant is really necessary. I added it as a replacement for the existing `JETPACK_SYNC_IDC_OPTIN` constant, but it seems like the filter (see below) should be sufficient.

Reply:
I guess a constant makes it easier to disable IDC if needed, without having to define a custom filter.

Reply:
I like the versatility of both a constant and a filter. However looking at the code wow its complicated to deprecate all the things. Sorry for the original suggestion of the name change